core/src/apps/cardano/certificates.py

```diff
@@ -0,0 +1,216 @@
+def validate_certificate(certificate: List[CardanoTxCertificateType]) -> None:
+    if certificate.type in [
+        CardanoCertificateType.STAKE_DELEGATION,
+        CardanoCertificateType.STAKE_REGISTRATION,
+        CardanoCertificateType.STAKE_DEREGISTRATION,
+    ]:
+        if not is_staking_path(certificate.path):
+            raise INVALID_CERTIFICATE
+
+    if certificate.type == CardanoCertificateType.STAKE_DELEGATION:
+        if not certificate.pool or len(certificate.pool) != POOL_HASH_SIZE:
+            raise INVALID_CERTIFICATE
+
+    if certificate.type == CardanoCertificateType.STAKEPOOL_REGISTRATION:
+        _validate_pool_parameters(certificate.pool_parameters)
+
+
+def cborize_certificate(
+    keychain: seed.Keychain,
+    certificate: CardanoTxCertificateType,
+    network_id: int,
+) -> Tuple:
+    if certificate.type in [
+        CardanoCertificateType.STAKE_REGISTRATION,
+        CardanoCertificateType.STAKE_DEREGISTRATION,
+    ]:
+        return (
+            certificate.type,
+            [0, get_public_key_hash(keychain, certificate.path)],
+        )
+    elif certificate.type == CardanoCertificateType.STAKE_DELEGATION:
+        return (
+            certificate.type,
+            [0, get_public_key_hash(keychain, certificate.path)],
+            certificate.pool,
+        )
+    elif certificate.type == CardanoCertificateType.STAKEPOOL_REGISTRATION:
+        pool_parameters = certificate.pool_parameters
+
+        return (
+            certificate.type,
+            pool_parameters.pool,
+            pool_parameters.vrf_key_hash,
+            pool_parameters.pledge,
+            pool_parameters.cost,
+            cbor.Tagged(
+                30,
+                (
+                    pool_parameters.margin_numerator,
+                    pool_parameters.margin_denominator,
+                ),
+            ),
+            pack_reward_address_bytes(
+                pool_parameters.reward_account_key_hash, network_id
+            ),
+            _cborize_pool_owners(keychain, pool_parameters.owners),
+            _cborize_pool_relays(pool_parameters.relays),
+            _cborize_pool_metadata(pool_parameters.metadata),
+        )
+    else:
+        raise INVALID_CERTIFICATE
+
+
+def _validate_pool_parameters(pool_parameters: CardanoPoolParametersType):
+    if any(
+        [
+            not pool_parameters.pool or len(pool_parameters.pool) != POOL_HASH_SIZE,
+            (
+                not pool_parameters.vrf_key_hash
+                or len(pool_parameters.vrf_key_hash) != VRF_KEY_HASH_SIZE
+            ),
+            not 0 <= pool_parameters.pledge < LOVELACE_MAX_SUPPLY,
+            not 0 <= pool_parameters.cost < LOVELACE_MAX_SUPPLY,
+            pool_parameters.margin_numerator <= 0,
+            pool_parameters.margin_denominator <= 0,
+            pool_parameters.margin_numerator > pool_parameters.margin_denominator,
+            (not pool_parameters.owners or len(pool_parameters.owners) == 0),
```

Comment:
Ledger allows only 1000 owners and relays (similarly for inputs, outputs etc.). The main reason (imho) is that the numbers are passed in APDUs, so we are essentially checking if the serialization format is correct (sending 1000 owner APDUs would take ages to confirm). So I agree such limits are not useful here and we do not need consistency between ledger and trezor.